src/web_peer_connection.cpp

```diff
@@ -86,16 +86,16 @@ web_peer_connection::web_peer_connection(peer_connection_args const& pack
 	shared_ptr<torrent> tor = pack.tor.lock();
 	TORRENT_ASSERT(tor);
 
-	// we always prefer downloading 1 MiB chunks
-	// from web seeds, or whole pieces if pieces
-	// are larger than a MiB
-	int preferred_size = 1024 * 1024;
+	// if the web server is known not to support keep-alive. request 4MiB
+	// but we want to have at least piece size to prevent block based requests
+	int const min_size = std::max((web.supports_keepalive ? 1 : 4) * 1024 * 1024,
```

Comment:
sorry for not thinking this all the way through in my earlier comments, but having these constants here (1 MiB and 4 MiB) seems a bit like a smell, now that it's configurable. The thing that makes sense in my mind is the logic that uses larger contiguous blocks if the web server does not support keep alive.

The only true lower limit here is 1 (which used to be expressed explicitly before this patch). I think that was a bit clearer than now, where it's expressed indirectly by making the piece_size a minimum request size, so when divided by the block size, it's never less than 1.

What would you think about something like this?

```
int const preferred_size = (web.supports_keepalive ? 1 : 2)
    * m_settings.get_int(settings_pack::urlseed_max_request_bytes);
prefer_contiguous_blocks(std::max(preferred_size / tor->block_size(), 1));
```
?

and would it make sense to call the setting ``settings_pack::urlseed_request_bytes``? or ``urlseed_request_size``? the word "max" to me implies that something else determines the size, and this is just a cap.